pkg/controllermanager/controller/secretbinding/secretbinding.go

```diff
@@ -99,20 +126,38 @@ func (c *Controller) Run(ctx context.Context, workers int) {
 
 	logger.Logger.Info("SecretBinding controller initialized.")
 
-	for i := 0; i < workers; i++ {
+	for i := 0; i < secretBindingWorkers; i++ {
 		controllerutils.CreateWorker(ctx, c.secretBindingQueue, "SecretBinding", c.reconciler, &waitGroup, c.workerCh)
 	}
+	if secretBindingProviderPopulatorEnabled {
+		for i := 0; i < secretBindingProviderPopulatorWorkers; i++ {
+			controllerutils.CreateWorker(ctx, c.shootQueue, "SecretBinding Provider Populator", c.secretBindingProviderPopulatorReconciler, &waitGroup, c.workerCh)
+		}
+	}
 
 	// Shutdown handling
 	<-ctx.Done()
 	c.secretBindingQueue.ShutDown()
+	if secretBindingProviderPopulatorEnabled {
+		c.shootQueue.ShutDown()
+	}
+
+	var (
+		secretBindingQueueLength = c.secretBindingQueue.Len()
+		queueLengths             = secretBindingQueueLength
+	)
+
+	if secretBindingProviderPopulatorEnabled {
+		var shootQueueLength = c.shootQueue.Len()
+		queueLengths += shootQueueLength
+	}
```

Comment:
Could be written more concisely by avoiding short lived variables that are only used once.
```suggestion
	queueLengths :=  c.secretBindingQueue.Len()
	if secretBindingProviderPopulatorEnabled {
		queueLengths += c.shootQueue.Len()
	}
```